presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAvroSchemaUrl.java

```diff
@@ -0,0 +1,101 @@
+import io.prestodb.tempto.ProductTest;
+import io.prestodb.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.InputStream;
+
+import static com.facebook.presto.tests.TestGroups.AVRO;
+import static com.facebook.presto.tests.TestGroups.HIVE_CONNECTOR;
+import static com.facebook.presto.tests.TestGroups.STORAGE_FORMATS;
+import static com.facebook.presto.tests.utils.QueryExecutors.onHive;
+import static com.facebook.presto.tests.utils.QueryExecutors.onPresto;
+import static io.prestodb.tempto.assertions.QueryAssert.Row.row;
+import static io.prestodb.tempto.assertions.QueryAssert.assertThat;
+import static java.lang.String.format;
+
+public class TestAvroSchemaUrl
+        extends ProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @BeforeTestWithContext
+    public void setup()
+            throws Exception
+    {
+        hdfsClient.createDirectory("/user/hive/warehouse/TestAvroSchemaUrl/schemas");
+        try (InputStream inputStream = Resources.asByteSource(Resources.getResource("avro/original_schema.avsc")).openStream()) {
+            hdfsClient.saveFile("/user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc", inputStream);
+        }
+    }
+
+    @AfterTestWithContext
+    public void cleanup()
+    {
+        hdfsClient.delete("/user/hive/warehouse/TestAvroSchemaUrl/schemas");
+    }
+
+    @DataProvider
+    public Object[][] avroSchemaLocations()
+    {
+        return new Object[][] {
+                {"file:///docker/volumes/presto-product-tests/avro/original_schema.avsc"}, // mounted in hadoop and presto containers
+                {"hdfs://hadoop-master:9000/user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc"},
+                // TODO {"hdfs:///user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc"},
+                // TODO {"/user/hive/warehouse/TestAvroSchemaUrl/schemas/original_schema.avsc"}, // `avro.schema.url` can actually be path on HDFS (not URL)
+        };
+    }
+
+    @Test(dataProvider = "avroSchemaLocations", groups = {AVRO, HIVE_CONNECTOR})
+    public void testHiveCreatedTable(String schemaLocation)
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS test_avro_schema_url_hive");
+        onHive().executeQuery(format("" +
+                        "CREATE TABLE test_avro_schema_url_hive " +
+                        "ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe' " +
+                        "STORED AS " +
+                        "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat' " +
+                        "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat' " +
+                        "TBLPROPERTIES ('avro.schema.url'='%s')",
+                schemaLocation));
+        onHive().executeQuery("INSERT INTO test_avro_schema_url_hive VALUES ('some text', 123042)");
```

Comment:
This is slow on Hive. This improves coverage, but maybe it's better not to have this? `testPrestoCreatedTable` remains

Reply:
i don't know

Reply:
> Is that because we have disable TEZ in docker images?

Yes. I am not sure if having TEZ enabled would even work in travis. We could give it a try.

Reply:
i don't want to open this potentially pandora box and going into Xmx settings between different processes on Travis